samplelistener/listener/dispatcher_service.bal

```diff
@@ -2,46 +2,45 @@ import ballerina/http;
 import ballerina/jballerina.java;
 
 service class DispatcherService {
-   private SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef;
-   //private string verificationToken;
+     private map<GenericService> services;
 
-   isolated function init(SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef) returns error? { //Receive verification token
-        self.serviceRef = serviceRef;
-        //self.verificationToken = "9asdbas9009123nas1e2";
+   isolated function init() returns error? {
+        self.services = {};
    }
 
-   isolated resource function post events (http:Caller caller, http:Request request) returns error? {
-        json payload = check request.getJsonPayload();
-        // string eventOrVerification = check payload.'type;
-
-        // if (payload.token !== self.verificationToken) {
-        //     return error("Verification token mismatch");
-        // }
+   isolated function addServiceRef(string serviceType, GenericService genericService) {
+        self.services[serviceType] = genericService;
+   }
 
-        // if (eventOrVerification == URL_VERIFICATION) {
-        //     check self.verifyURL(caller, payload);
-        // } else if (eventOrVerification == EVENT_CALLBACK) {
-        // }
+   resource function post events (http:Caller caller, http:Request request) returns error? {
+        json payload = check request.getJsonPayload();
 
         GenericEventWrapperEvent genericEvent = check payload.cloneWithType(GenericEventWrapperEvent);
-        if (genericEvent.event.'type == "app_mention") {
-                SlackAppMentionHandlingService serviceReference = <SlackAppMentionHandlingService> self.serviceRef;
-                var s = check self.callOnAppEvent(genericEvent, "app_mention", "onAppMention", serviceReference);
+        match genericEvent.event.'type {
+             "app_mention_added" => {
+                if self.services.hasKey("AppMentionHandlingService") {
```

Comment:
IMO, better to use constants for these values.

Also, the keys are something we define, right? Can't we use the event type itself as the key?

Reply:
Since this is going to be a generated code, should we use a separate constants file to store the constants?

A service can be related to multiple events, as in L22 and L27, the same key is being used, but the event type is different. Hence IMO, we can't use the event type itself as the key.

Reply:
> Since this is going to be a generated code, do we need to use a separate constants file to store the constants in a generated code?
> 


They can go in any file.

> A service can be related to multiple events, as in L22 and L27, the same key is being used, but the event type is different. Hence IMO, we can't use the event type itself as the key.


Ah yes, forgot about this.

Reply:
Resolved